### PR TITLE
Add download-latest release support

### PR DESCRIPTION
**Pull request by PerfectSlayer**
Unlike the download-latest-vX, the download-latest release is not bound to any major version and will be updated to any newer version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,8 +4,8 @@ This GitHub action updates meta releases, called _download releases_, to always
 
 ## Principle
 
-The action scans for published non-draft releases with tags `vX.Y.Z`, find the latest for each major `X` version, and ensure the related _download release_ tagged `download-latest_vX` refers to the latest version.
-If not, it downloads the `dd-java-agent.jar `attachment from the latest version, upload it to the _download release_ and update its body to refers to the latest version.
+The action scans for published non-draft releases with tags `vX.Y.Z`, find the latest for each major `X` version, and ensure the related _download releases_ tagged `download-latest_vX` and `download-latest` refer to their latest version.
```
Comment (bantonsson) [inline README.md:7]: _... refer to their latest versions._

File: README.md [hunk 2]
```diff
@@ -4,8 +4,8 @@ This GitHub action updates meta releases, called _download releases_, to always
...
+If not, it downloads the `dd-java-agent.jar `attachment from the latest version, upload it to the _download releases_ and update theirs bodies to refer to the latest version.
```
Comment (bantonsson) [inline README.md:8]: _... update their bodies to refer to their latest versions._

File: src/types.ts [hunk 3]
```diff
@@ -57,15 +58,16 @@ class DownloadRelease {
   }
 
   static fromTag(id: number, tag: String): DownloadRelease | undefined {
-    const pattern = /download-latest-v(\d+)/
+    const pattern = /^download-latest(?:-v(\d+))?$/
```
Comment (bantonsson) [inline src/types.ts:61]: Is the `(?:...` really necessary since you have a `?` after the outer group?

**Review by bantonsson — APPROVED** [review]
Only minor nitpicks

File: src/types.ts (hunk 3 above)
Comment (PerfectSlayer) [reply src/types.ts:61]: The `?` of `(?:` is not a quantifier but an option to turn the `(` as a non capturing group.
As I don't care about capturing `-vX` but just only `X`, I disabled it (by I had to have some parenthesis to define the scope of the second `?`).
Am I clear? (I does not sound while I am typing it 😓 )

File: README.md (hunk 2 above)
Comment (PerfectSlayer) [reply README.md:8]: My bad, thanks for the review!

File: src/types.ts (hunk 3 above)
Comment (bantonsson) [reply src/types.ts:61]: Ah of course. That makes sense. Thanks for the explanation.

**Pull request merged**